Yoast/wordpress-seo
# Bugreports 633 notification re index is never shown

## Summary


This PR can be summarized in the following changelog entry:

* [wordpress-seo-premium] Fixes a bug where the notification to analyze your content for internal linking suggestions was not shown. 

## Relevant technical choices:

* Sets a `user` option on the notification and uses this stored user to check the capabilities against. 
  * Defaults to the current user.

## Test instructions

This PR can be tested by following these steps:

This PR needs to be tested in tandem with this premium PR: https://github.com/Yoast/wordpress-seo-premium/pull/2603
See the premium PR for test instructions.

## UI changes

## Documentation

## Quality assurance


Fixes https://github.com/Yoast/bugreports/issues/633
